### PR TITLE
Refactor quantization configs usage

### PR DESCRIPTION
**Pull request by reuvenperetz**

### CODE REVIEW

File: model_compression_toolkit/common/collectors/min_max_per_channel_collector.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class MinMaxPerChannelCollector(BaseCollector):
     def __init__(self,
                  init_min_value: float = None,
                  init_max_value: float = None,
-                 axis=-1):
+                 axis: int = None):
```
Comment (haihabi) [inline model_compression_toolkit/common/collectors/min_max_per_channel_collector.py:29]: If the axis is None, there should be an exception ?

File: model_compression_toolkit/common/base_node_prior_info.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from model_compression_toolkit.common.framework_info import FrameworkInfo
+from model_compression_toolkit.common import BaseNode
+
+
+class BaseNodePriorInfo:
+
+    def __init__(self,
+                 node: BaseNode,
+                 fw_info: FrameworkInfo):
```
Comment (haihabi) [inline model_compression_toolkit/common/base_node_prior_info.py:9]: Why do we need Framework Info

File: model_compression_toolkit/common/collectors/statistics_collector.py [hunk 3]
```diff
@@ -53,14 +50,15 @@ def update_statistics(self,
...
+class StatsCollector(BaseStatsCollector):
     """
     Class to wrap all statistics that are being collected for an input/output node.
     """
 
     def __init__(self,
                  init_min_value: float = None,
-                 init_max_value: float = None):
+                 init_max_value: float = None,
+                 output_channel_index: int = None):
```
Comment (haihabi) [inline model_compression_toolkit/common/collectors/statistics_collector.py:61]: If the axis is None, there should be an exception ?

File: model_compression_toolkit/common/framework_info.py [hunk 4]
```diff
@@ -34,7 +34,8 @@ def __init__(self,
                  kernel_channels_mapping: DefaultDict,
                  activation_min_max_mapping: Dict[str, tuple],
                  layer_min_max_mapping: Dict[Any, tuple],
-                 kernel_ops_attributes_mapping: DefaultDict):
+                 kernel_ops_attributes_mapping: DefaultDict,
+                 output_channel_index: int):
```
Comment (haihabi) [inline model_compression_toolkit/common/framework_info.py:38]: Add comments

File: model_compression_toolkit/common/graph/base_graph.py [hunk 5]
```diff
@@ -418,30 +416,34 @@ def get_memory(self) -> float:
...
         """
         Get a list of nodes that can be configured (namely, has one or
-        more weight qc candidate). The nodes are sorted according to the topological
-        order of the graph.
+        more weight qc candidate and their weights should be quantized).
+        The nodes are sorted according to the topological order of the graph.
 
         Args:
             include_reused_nodes: Whether or not to include reused nodes (False by default).
+            fw_info: Information needed for quantization about the specific framework.
```
Comment (haihabi) [inline model_compression_toolkit/common/graph/base_graph.py:446]: Remove fw_info from Args commetns

File: model_compression_toolkit/common/mixed_precision/bit_width_setter.py [hunk 6]
```diff
@@ -54,7 +54,7 @@ def set_bit_widths(quant_config: QuantizationConfig,
         # only candidate of the node as its final weight quantization configuration.
         if len(quant_config.weights_n_bits) == 1:
             for n in graph.nodes:
-                if n.name in graph.get_configurable_sorted_nodes_names():
+                if n.name in graph.get_configurable_sorted_nodes_names(fw_info=fw_info):
```
Comment (haihabi) [inline model_compression_toolkit/common/mixed_precision/bit_width_setter.py:57]: Why does the function get_configurable_sorted_nodes_names get fw_info as input?

File: model_compression_toolkit/keras/keras_implementation.py [hunk 7]
```diff
@@ -116,23 +118,24 @@ def shift_negative_correction(self,
...
-                          fw_info: FrameworkInfo) -> common.statistics_collector.BaseStatsContainer:
+    def attach_sc_to_node(self,
+                          node: BaseNode,
+                          output_channel_index: int) -> BaseStatsCollector:
         """
         Return a statistics collector that should be attached to a node's output
         during statistics collection.
 
         Args:
             node: Node to return its collector.
```
Comment (haihabi) [inline model_compression_toolkit/keras/keras_implementation.py:132]: missing argument commets

File: model_compression_toolkit/keras/keras_node_prior_info.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+from model_compression_toolkit.keras.constants import ACTIVATION, RELU_MAX_VALUE, NEGATIVE_SLOPE, THRESHOLD
+
+
+class KerasNodePriorInfo(BaseNodePriorInfo):
+
+    def __init__(self,
+                 node: BaseNode,
+                 fw_info: FrameworkInfo):
+        super(KerasNodePriorInfo, self).__init__(node=node,
+                                                 fw_info=fw_info)
```
Comment (haihabi) [inline model_compression_toolkit/keras/keras_node_prior_info.py:22]: Why prior Info class is not just a container?
